solr/core/src/test/org/apache/solr/cloud/CollectionsAPIDistributedZkTest.java

```diff
@@ -435,73 +278,55 @@ private void testErrorHandling() throws Exception {
     request.setPath("/admin/collections");
     gotExp = false;
     try {
-      makeRequest(baseUrl, request);
+      cloudClient.request(request);
     } catch (SolrException e) {
       gotExp = true;
     }
     assertTrue(gotExp);
     
-    // Too many replicas
+    // No numShards should fail
     params = new ModifiableSolrParams();
     params.set("action", CollectionAction.CREATE.toString());
-    collectionName = "collection";
+    collectionName = "acollection";
     params.set("name", collectionName);
-    params.set("numShards", 2);
+    params.set(REPLICATION_FACTOR, 10);
     if (secondConfigSet) {
       params.set("collection.configName", "conf1");
     }
-    params.set(REPLICATION_FACTOR, 10);
     request = new QueryRequest(params);
     request.setPath("/admin/collections");
     gotExp = false;
     try {
-      makeRequest(baseUrl, request);
+      cloudClient.request(request);
     } catch (SolrException e) {
       gotExp = true;
     }
     assertTrue(gotExp);
+  }
+  
+  @Test
+  public void testBadParams() throws SolrServerException, IOException {
+    int exceptionCount = 0;
     
-    // No numShards should fail
-    params = new ModifiableSolrParams();
-    params.set("action", CollectionAction.CREATE.toString());
-    collectionName = "acollection";
-    params.set("name", collectionName);
-    params.set(REPLICATION_FACTOR, 10);
-    if (secondConfigSet) {
-      params.set("collection.configName", "conf1");
-    }
-    request = new QueryRequest(params);
-    request.setPath("/admin/collections");
-    gotExp = false;
+    // Too many replicas
     try {
-      makeRequest(baseUrl, request);
+      createCollection("collection", configSet, 2, 10).process(cloudClient);
     } catch (SolrException e) {
-      gotExp = true;
+      ++exceptionCount;
```

Comment:
Rather than doing this, can you use ```expectThrows()```?  That way we get better error messages if the test fails.